common/src/main/java/org/apache/cassandra/sidecar/common/NodeSettings.java

```diff
@@ -16,33 +16,82 @@
  * limitations under the License.
  */
 
-
 package org.apache.cassandra.sidecar.common;
 
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Holds information about the specific node settings
  */
 public class NodeSettings
 {
-    private final String releaseVersion;
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeSettings.class);
+    private static final String SIDECAR_VERSION = getSidecarVersion();
+
     private final String partitioner;
+    private final String releaseVersion;
+    private final String sidecarVersion;
+
+    private static String getSidecarVersion()
+    {
+        try (InputStream input = NodeSettings.class.getResourceAsStream("/sidecar.version");
```

Comment:
maybe define the string as a constant?

Reply:
I don't think having a constant here will help, as it cannot be referenced from the Gradle build file… Local variable, maybe?

Reply:
Added a local final variable.